src/EnergyPlus/HWBaseboardRadiator.cc

```diff
@@ -664,74 +613,73 @@ namespace HWBaseboardRadiator {
                                         cCMO_BBRadiator_Water,
                                         state.dataIPShortCut->cAlphaArgs(1)));
                 HWBaseboardDesignDataObject.FracRadiant = MaxFraction;
-                HWBaseboard(BaseboardNum).FracConvect = 0.0;
+                thisHWBaseboard.FracConvect = 0.0;
             } else {
-                HWBaseboard(BaseboardNum).FracConvect = 1.0 - AllFracsSummed;
+                thisHWBaseboard.FracConvect = 1.0 - AllFracsSummed;
             }
 
-            HWBaseboard(BaseboardNum).TotSurfToDistrib = NumNumbers - 4;
-            //      IF (HWBaseboard(BaseboardNum)%TotSurfToDistrib > MaxDistribSurfaces) THEN
+            thisHWBaseboard.TotSurfToDistrib = NumNumbers - 4;
+            //      IF (thisHWBaseboard%TotSurfToDistrib > MaxDistribSurfaces) THEN
             //        CALL ShowWarningError(state, RoutineName//cCMO_BBRadiator_Water//'="'//TRIM(state.dataIPShortCut->cAlphaArgs(1))// &
             //          '", the number of surface/radiant fraction groups entered was higher than the allowable maximum.')
             //        CALL ShowContinueError(state, '...only the maximum value=['//TRIM(RoundSigDigits(MaxDistribSurfaces))// &
             //           '] will be processed.')
-            //        HWBaseboard(BaseboardNum)%TotSurfToDistrib = MaxDistribSurfaces
+            //        thisHWBaseboard%TotSurfToDistrib = MaxDistribSurfaces
             //      END IF
-            if ((HWBaseboard(BaseboardNum).TotSurfToDistrib < MinDistribSurfaces) && (HWBaseboardDesignDataObject.FracRadiant > MinFraction)) {
+            if ((thisHWBaseboard.TotSurfToDistrib < MinDistribSurfaces) && (HWBaseboardDesignDataObject.FracRadiant > MinFraction)) {
                 ShowSevereError(state,
                                 std::string{RoutineName} + cCMO_BBRadiator_Water + "=\"" + state.dataIPShortCut->cAlphaArgs(1) +
                                     "\", the number of surface/radiant fraction groups entered was less than the allowable minimum.");
                 ShowContinueError(state, format("...the minimum that must be entered=[{}].", MinDistribSurfaces));
                 ErrorsFound = true;
-                HWBaseboard(BaseboardNum).TotSurfToDistrib = 0; // error
+                thisHWBaseboard.TotSurfToDistrib = 0; // error
             }
 
-            HWBaseboard(BaseboardNum).SurfaceName.allocate(HWBaseboard(BaseboardNum).TotSurfToDistrib);
-            HWBaseboard(BaseboardNum).SurfaceName = "";
-            HWBaseboard(BaseboardNum).SurfacePtr.allocate(HWBaseboard(BaseboardNum).TotSurfToDistrib);
-            HWBaseboard(BaseboardNum).SurfacePtr = 0;
-            HWBaseboard(BaseboardNum).FracDistribToSurf.allocate(HWBaseboard(BaseboardNum).TotSurfToDistrib);
-            HWBaseboard(BaseboardNum).FracDistribToSurf = 0.0;
+            thisHWBaseboard.SurfaceName.allocate(thisHWBaseboard.TotSurfToDistrib);
```

Comment:
Do you really need to save the surface names if you are immediately converting them to surface indices below?